tests/TestCase/ORM/QueryTest.php

```diff
@@ -1328,13 +1344,17 @@ public function testFormatResultsBelongsToMany()
             $this->assertInstanceOf('Cake\ORM\Entity', $r);
         }
 
+        $this->assertTrue($first->beforeFind);
+        $this->assertNull($first->joinBeforeFind);
+
         $this->assertCount(2, $first->tags);
         $expected = [
             'id' => 1,
             'name' => 'tag1',
             '_joinData' => ['article_id' => 1, 'tag_id' => 1],
             'description' => 'A big description',
             'created' => new Time('2016-01-01 00:00'),
+            'joinBeforeFind' => true,
```

Comment:
Ok. I originally had it in _joinData (a much easier fix) but I wrote the original test case incorrectly and followed it. Oy what a waste. At any rate, I'll rebase.

I've got to check our app code to make sure the issue I originally discovered is even related to this one now :)